apps/builder/app/builder/features/style-panel/sections/advanced/parse-style-input.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  properties,
+  parseCss,
+  type ParsedStyleDecl,
+} from "@webstudio-is/css-data";
+import { type StyleProperty } from "@webstudio-is/css-engine";
+import { camelCase } from "change-case";
+import { lexer } from "css-tree";
+
+/**
+ * Does several attempts to parse:
+ * - Custom property "--foo"
+ * - Known regular property "color"
+ * - Custom property without -- (user forgot to add)
+ * - Custom property and value: --foo: red
+ * - Property and value: color: red
+ * - Multiple properties: color: red; background: blue
+ */
+export const parseStyleInput = (css: string): Array<ParsedStyleDecl> => {
+  css = css.trim();
+  // Is it a custom property "--foo"?
+  if (css.startsWith("--") && lexer.match("<custom-ident>", css).matched) {
+    return [
+      {
+        selector: "selector",
+        property: css as StyleProperty,
+        value: { type: "unset", value: "" },
+      },
+    ];
+  }
+
+  // Is it a known regular property?
+  const camelCasedProperty = camelCase(css);
+  if (camelCasedProperty in properties) {
+    return [
+      {
+        selector: "selector",
+        property: css as StyleProperty,
+        value: { type: "unset", value: "" },
+      },
+    ];
+  }
```

Comment:
isn't this what parseCss does?

Reply:
Yes, but the parser will also let non-existing properties pass through and it will also not convert "bla" to "--bla" which is something I discovered people may want as they don't know  of forget that a variable needs "--" when adding property in advanced.

I could add all that logic directly in to parseCss, but this is so biased towards how UX in advanced panel needs to be that I am not sure this is a good idea to make parseCss behave in such a special way

Reply:
Idk, maybe its ok to make our parseCss behave this way? wdyt?
basically all it needs is to not let unknown properties get through and instead convert them to custom properties by adding dashes

Reply:
no, blablubb is going to be a valid proeprty, I am literally fixing this right now with this PR as reported by John

Reply:
Are you sure it does not output `{ type: 'invalid', value: input }`? Maybe need to filter by type before checking length.

Reply:
yes, that's why I had to write this function

Reply:
we could make the parseCss though check the list of properties with our list, so it would return type invalid, then I could wrap that functionality into that specific function I have now with additional UX for auto variables conversion

Reply:
Maybe just filter out invalid properties inside of parseCss and go with previous approach?

Reply:
invalid as in missing in our list of known properties?

Reply:
I wanted to still know which once are invalid and provide an option to the user create a custom property out of them

Reply:
when its just one property provided by user and empty array returned from parser - this is easy to assume there was nothing valid etc, but then maybe its still better to have a way to know when there are multiple properties that are invalid and make them all automatically into custom properties, this way we don't have to completely ignore them